src/routes/data.js

```diff
@@ -0,0 +1,240 @@
+      });
+
+      if (result.length === 0) {
+        return res.status(204).end();
+      }
+
+      res.type('application/json');
+      res.json(result);
+    } catch (error) {
+      log.error(error.message);
+      res.status(500).send(error.message);
+    }
+  }
+);
+/**
+ * PUT Update a Diluent
+ * @param id int
+ * @body name string
+ * @body slug string
+ * @body code string
+ * @body density numeric
+ */
+router.put(
+  '/supplier/:id',
+  authenticate(),
+  [
+    param('id')
+      .isNumeric()
+      .isInt({ min: 1 })
+      .toInt(),
+    body('name')
+      .isString()
+      .isLength({ min: 1 })
+      .withMessage('length'),
+    body('code').isString()
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    log.info(`request to update data supplier id ${req.params.id}`);
+    try {
+      const result = await DataSupplier.update(
+        {
+          name: req.body.name,
+          code: req.body.code
+        },
+        {
+          where: {
+            id: req.params.id
+          }
+        }
+      );
+
+      if (result.length === 0) {
+        return res.status(204).end();
+      }
+
+      res.type('application/json');
+      res.json(result);
+    } catch (error) {
+      log.error(error.message);
+      res.status(500).send(error.message);
+    }
+  }
+);
+/**
+ * Delete Diluent
+ * @param id int
+ */
+router.delete(
+  '/supplier/:id',
+  authenticate(),
+  [
+    param('id')
+      .isNumeric()
+      .isInt({ min: 1 })
+      .toInt()
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    log.info(`request to delete data supplier id ${req.params.id}`);
+    try {
+      const result = await DataSupplier.destroy({
+        where: {
+          id: req.params.id
+        }
+      });
+
+      if (!result || result.length === 0) {
+        return res.status(204).end();
+      }
+
+      res.type('application/json');
+      res.json(result);
+    } catch (error) {
+      log.error(error.message);
+      res.status(500).send(error.message);
+    }
+  }
+);
+/**
+ * GET Data Suppliers
+ */
+router.get('/suppliers', authenticate(), async (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
+
+  log.info(`request for data suppliers`);
+  try {
+    const result = await DataSupplier.findAll();
+
+    if (!result || result.length === 0) {
+      return res.status(204).end();
+    }
+
+    res.type('application/json');
+    res.json(result);
+  } catch (error) {
+    log.error(error.message);
+    res.status(500).send(error.message);
+  }
+});
+/**
+ * GET Database Schema Version Info
+ */
+router.get('/version', authenticate(), async (req, res) => {
```

Comment:
This is interesting - I don't think we should give out all of this info to anyone with a token. Is there a use case you have in mind?

Reply:
It should be behind an admin role check, once that is in place. I just thought it would be useful to be able to check the current migration state without having direct access to the database.